File: leetcode/medium/1503. Last Moment Before All Ants Fall Out of a Plank/main.go
package main

func getLastMoment(n int, left []int, right []int) int {
	maxT := 0

	for _, v := range right {
		maxT = max(maxT, n-v)
	}

	for _, v := range left {
		maxT = max(maxT, v)
	}

	return maxT
}

func max(a, b int) int {
	if a > b {
		return a
	}

	return b
}

File: leetcode/medium/1503. Last Moment Before All Ants Fall Out of a Plank/main.ts
const getLastMoment = (n: number, left: number[], right: number[]): number => {
  let result = 0

  left.forEach((v, _) => (result = Math.max(result, v)))
  right.forEach((v, _) => (result = Math.max(result, n - v)))

  return result
}
